ixmp/testing/__init__.py

```diff
@@ -183,6 +188,40 @@ def pytest_sessionstart(session: pytest.Session) -> None:
 
     jdbc._GC_AGGRESSIVE = False
 
+    if "ixmp4" in available():
+        from sqlalchemy import create_engine, text
+        from xdist import get_xdist_worker_id
+
+        db_name = f"ixmp_test_{get_xdist_worker_id(session)}"
+
+        engine = create_engine(
+            "postgresql+psycopg://postgres:postgres@localhost:5432/postgres",
+            isolation_level="AUTOCOMMIT",
+        )
+        with engine.connect() as connection:
+            connection.execute(text(f"CREATE DATABASE {db_name}"))
+
+        engine.dispose()
+
+
+def pytest_sessionfinish(
+    session: pytest.Session, exitstatus: int | pytest.ExitCode
+) -> None:
+    if "ixmp4" in available():
+        from sqlalchemy import create_engine, text
+        from xdist import get_xdist_worker_id
+
+        db_name = f"ixmp_test_{get_xdist_worker_id(session)}"
+
+        engine = create_engine(
```

Comment:
Instead of having very similar code to above, the [`pytestconfig.stash`](https://docs.pytest.org/en/stable/reference/reference.html#pytest.Config.stash) feature can be used.

Reply:
> So I would also need to store this key somehow for later usage.

Yes, that's the intended usage. Maybe https://docs.pytest.org/en/stable/example/simple.html#making-test-result-information-available-in-fixtures is a more direct example in the pytest docs.

See also https://github.com/khaeru/sdmx/blob/abf1d4301decb645cfe4bfac926c5a4f18150470/sdmx/testing/__init__.py#L33-L36